lib/active_model/serializer/type.rb

```diff
@@ -16,8 +16,17 @@ module ClassMethods
         # @example
         #   class AdminAuthorSerializer < ActiveModel::Serializer
         #     type 'authors'
-        def type(type)
-          self._type = type && type.to_s
+        #
+        # @example
+        #   class AdminAuthorSerializer < ActiveModel::Serializer
+        #     type { |object| object.class.name }
+        #
+        # @example
+        #   class AdminAuthorSerializer < ActiveModel::Serializer
+        #     type proc { |object| object.class.name }
+        #
+        def type(type = nil, &block)
+          self._type = block || type
```

Comment:
lol. maybe we should figure out the bigger problem first of json_key, root, _type? :-)
I'll update / redo this PR when we figure all that out.